src/lib.rs

```diff
@@ -14,5 +14,4 @@ pub type CAMetalLayer = *mut Object;
 pub enum Layer {
     Existing(CAMetalLayer),
     Allocated(CAMetalLayer),
-    None,
 }
```

Comment:
In `ash-window` we have a pattern that takes out the `CAMetalLayer`. Is there any significance here (e.g. lifetimes) or can we flatten this, store existing vs allocated in a separate member, or add a helper `fn value(self) -> CAMetalLayer` thing?

Reply:
The `CAMetalLayer` may be allocated in case there couldn't be extracted one from the window handle, and should be destroyed if not needed anymore (on surface destruction). To not leak the consumer should execute sth like `msg_send![layer, release]` in case of `Layer::Allocated(layer)` (but couldn't test due to lack of hw)

Reply:
Since `raw-window-metal` maps this nicely, should it also implement `Drop` that does this for `Allocated` but not for `Existing`? Or at least add a doc-comment and perhaps make this an "implementation detail" unless transferring ownership is a thing, which we should make more explicit to the caller?

Reply:
I could test these things out soon™, perhaps worth doing that before cutting a new release?